src/shaders/_prelude.fragment.glsl

```diff
@@ -17,3 +17,40 @@ precision mediump float;
 #endif
 
 const float PI = 3.141592653589793;
+
+// Comment to globally disable gamma correction (presently only wrt fog):
+#define GAMMA_CORRECT
+
+vec3 linearToSRGB(vec3 color) {
+// TODO: Make a choice and remove this conditional before production
+#ifdef GAMMA_CORRECT
+    return pow(color, vec3(1.0 / 2.2));
+    /*
+    vec3 a = 12.92 * color;
+    vec3 b = 1.055 * pow(color, vec3(1.0 / 2.4)) - 0.055;
+    vec3 c = step(vec3(0.0031308), color);
+    return mix(a, b, c);
+    */
+#else
+    return color;
+#endif
+}
+
+vec3 srgbToLinear(vec3 color) {
+// TODO: Make a choice and remove this conditional before production
+#ifdef GAMMA_CORRECT
+    return pow(color, vec3(2.2));
+    /*
+    vec3 a = color / 12.92;
+    vec3 b = pow((color + 0.055) / 1.055, vec3(2.4));
+    vec3 c = step(vec3(0.04045), color);
+    return mix(a, b, c);
+    */
+#else
+    return color;
+#endif
+}
+
+vec3 gammaMix(vec3 a, vec3 b, float x) {
```

Comment:
I'm in favor of leaving it gamma correct. The best test case is black fog on a white style, making these transitions a lot darker than they should be:

**without GAMMA_CORRECT**
![1](https://user-images.githubusercontent.com/7061573/112181531-c37ea680-8bb9-11eb-80fc-0a9eb4155ffd.png)
**with GAMMA_CORRECT**
![2](https://user-images.githubusercontent.com/7061573/112181518-bfeb1f80-8bb9-11eb-8427-e051b450bd67.png)

Which acts exactly as the issue we could see with a simple gradient:
<img width="393" alt="Screen Shot 2020-05-04 at 3 00 16 PM (1)" src="https://user-images.githubusercontent.com/7061573/112181972-207a5c80-8bba-11eb-91fb-d3947d268c46.png">
<img width="392" alt="Screen Shot 2020-05-04 at 3 08 28 PM (1)" src="https://user-images.githubusercontent.com/7061573/112181981-22442000-8bba-11eb-8d54-4bffedb230fd.png">

Concerning cost, the simple pow function are good enough here, I don't think we should use the more accurate versions (It's hard to notice a difference using them), and may only be a critical choice for complex lighting.

One nit: we don't have a linter in shader code, but it seems like snake_case is prevalent for function names/variables.

Reply:
+1 for adding gamma correction on the fog, the greater spread definitely adds to the effect for me.

Reply:
@karimnaaji in the black-to-white gradients above, is the top one gamma-incorrect and the bottom gamma-correct? The bottom one looks unreasonably light, on average.

Reply:
The medium part of second gradient is at (0.73,0.73,0.73) srgb, which is exactly (0.5,0.5,0.5) linear. 

<img width="371" alt="Screen Shot 2021-03-24 at 2 40 08 PM" src="https://user-images.githubusercontent.com/7061573/112387480-0e2e1a80-8caf-11eb-8df9-9f62760ba854.png">

I still had the shader around to generate these:
```
out vec4 out_color;
uniform vec2 resolution;

@color3
uniform vec3 color_1;
@color3
uniform vec3 color_2;

vec3 linear2srgb(vec3 color) {
    return pow(color, vec3(1.0 / 2.2));
}
vec3 srgb2linear(vec3 color) {
    return pow(color, vec3(2.2));
}
void main() {
    vec2 uv = gl_FragCoord.xy / resolution;

    if (uv.y >= 0.5) {
        out_color = vec4(1.0);
        vec3 color = mix(color_1, color_2, uv.x);
        out_color = vec4(color, 1.0);
    } else {
        vec3 color = mix(srgb2linear(color_1), srgb2linear(color_2), uv.x);
        out_color = vec4(linear2srgb(color), 1.0);
    }
}
```
The bottom one was  was generated with gamma correct mixing between black and white, which is exactly what we do with fog now.

While looking at that I remember comparing the heatmap as well, which is another place we'd benefit from the same process (top is gamma corrected):

<img width="721" alt="Screen Shot 2021-03-24 at 2 37 29 PM" src="https://user-images.githubusercontent.com/7061573/112386736-8f38e200-8cae-11eb-9fe3-a6a769f37380.png">
<img width="726" alt="Screen Shot 2021-03-24 at 2 37 46 PM" src="https://user-images.githubusercontent.com/7061573/112386738-8fd17880-8cae-11eb-9596-d62915902f55.png">

Reply:
The bottom one sort of looks like it has a dark ring where the opacity fades. This might be too much of a breaking change, but I do think it is objectively better by a small margin.